system_metrics_collector/CMakeLists.txt

```diff
@@ -41,11 +42,13 @@ add_library(system_metrics_collector SHARED
   src/moving_average_statistics/moving_average.cpp
   src/moving_average_statistics/types.hpp
 )
-
-ament_target_dependencies(system_metrics_collector rclcpp rcpputils)
+target_include_directories(system_metrics_collector PRIVATE ${metrics_statistics_msgs_INCLUDE_DIRS})
+target_link_libraries(system_metrics_collector ${metrics_statistics_msgs_LIBRARIES})
+ament_target_dependencies(system_metrics_collector rclcpp rcutils rcpputils metrics_statistics_msgs)
 ament_export_libraries(system_metrics_collector)
 
 add_executable(main src/system_metrics_collector/main.cpp)
+target_include_directories(main PRIVATE ${metrics_statistics_msgs_INCLUDE_DIRS})
```

Comment:
ditto - why is `ament_target_dependencies` not sufficient?

Reply:
Removed `target_include_directories()`.